alacritty_terminal/src/term/mod.rs

```diff
@@ -887,6 +896,8 @@ impl<T> Term<T> {
             clipboard,
             event_proxy,
             is_focused: true,
+            title: config.window.title.as_ref().map_or(DEFAULT_NAME.to_string(), |t| t.to_string()),
```

Comment:
We don't need to use `map_or` here, instead this is a bit more clear as to what's actually happening:

```suggestion
            title: config.window.title.clone().unwrap_or_else(|| DEFAULT_NAME.to_owned()),
```

However, I'd assume that we use this in window.rs too and that's where you got this code from, in which case we should really rather centralize it.

I'd recommend either adding a `title` method to our config, or to simply use serde to always deserialize to a `String` instead of `Option<String>` since it seems we do always need it anyways. So we can just use serde's defaults for this.

Reply:
One thing about changing `title` from `Option<String>` to just `String` is that the override for `dynamic_title` depends on checking for the value's existence. Here's a couple of options I can think of to resolve this:

- Check if the title is equal to the default value "Alacritty" instead. This seems a little bit like a magic string if you ask me.
- Keep it as an `Option<String>` and duplicate the logic (`unwrap_or_else` with some sort of utility function might work).
- Don't override `dynamic_title` based on the `title` config field being defined. This would be a breaking change, of course.

Do you have any suggestions?

Reply:
:+1: for centralizing the default. We will now have duplication between the window value `title` and the term value `title`, ideally we should only have one. Either the `Term` should populate the title by default with the value of the `Window`'s title, or the window should be wired up to use the `Term` default IMO. This isn't necessarily going to look super clean since the `Display` (which owns the `Window`) is pretty separate from the `Term`.

I'm wondering if it doesn't make sense to reuse the `Event::Title` to send the default from the `Term`. Just a thought...

Reply:
> We will now have duplication between the window value title and the term value title

This is incorrect, we do not store the title on the window.

> One thing about changing title from Option<String> to just String is that the override for dynamic_title depends on checking for the value's existence.

Seems like the simplest option is to just add a `title` method to the config then.

Reply:
@chrisduerr, oh your totally right. I think I was looking at the `WindowConfig`, which of course makes sense to have a title option.

Sorry for the confusion.

Reply:
> Seems like the simplest option is to just add a title method to the config then.

Usually we'd make the `title` field private in this case, and always access it with the `title()` method which still wouldn't indicate if it was defined in the config or not.

Reply:
> Usually we'd make the title field private in this case, and always access it with the title() method which still wouldn't indicate if it was defined in the config or not.

That sounds like what I had in mind, yeah.